pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -137,7 +138,15 @@ func Execute(configFile io.Reader) {
 	app.Config.HTTP.Headers.Set("X-Registry-Supports-Signatures", "1")
 
 	app.RegisterHealthChecks()
-	handler := alive("/", app)
+	handler := http.Handler(app)
```

Comment:
Any write (upload) should be part of the same rate limit pool.  Otherwise we have to come up with heuristics and split the pool, which then creates more operational overhead.